cmd/tsidp/tsidp.go

```diff
@@ -67,10 +73,12 @@ var (
 	flagFunnel             = flag.Bool("funnel", false, "use Tailscale Funnel to make tsidp available on the public internet")
 	flagHostname           = flag.String("hostname", "idp", "tsnet hostname to use instead of idp")
 	flagDir                = flag.String("dir", "", "tsnet state directory; a default one will be created if not provided")
+	flagState              = flag.String("state", os.Getenv("TS_STATE"), "path to tailscale state file or 'kube:<secret-name>' to use Kubernetes secret; if unset, 'dir' is used")
```

Comment:
let's omit the inconsistent `os.Getenv("TS_STATE")` bit and just default to the empty string. The Docker/Kubernetes stuff can use environment variables if it wants.